### PR TITLE
Disable buttons during API calls

### PR DESCRIPTION
**Pull request by maciaszczykm**
### Which issue(s) this PR fixes
Closes https://github.com/kubermatic/dashboard/issues/4264.

### Release note
```release-note
NONE
```

### CODE REVIEW

File: src/app/shared/components/button/component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@Component({
+  selector: 'km-button',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ButtonComponent implements OnInit, OnDestroy {
+  @Input() icon: string;
+  @Input() label: string;
+  @Input() observable: Observable<any>;
```
Comment (floreks) [inline src/app/shared/components/button/component.ts:28]: If generic approach (`T`) does not work, I'd use here `unknown` instead of `any`

**Comment by maciaszczykm** [comment]
/assign @floreks 

I have created https://github.com/kubermatic/dashboard/issues/4379 to finish replacement process. The base seems to be ready.

File: src/app/backup/list/restore/template.html [hunk 2]
```diff
@@ -99,8 +99,7 @@
                 <button mat-icon-button
                         [attr.id]="'km-delete-restore-' + backup.name"
                         matTooltip="Delete Automatic Backup"
-                        kmThrottleClick
-                        (throttleClick)="delete(backup); $event.stopPropagation()"
+                        (click)="delete(backup); $event.stopPropagation()"
```
Comment (maciaszczykm) [inline src/app/backup/list/restore/template.html:102]: It was not required as API is not called in the handler. It opens a dialog which doesn't need additional throttling.

File: src/app/shared/components/cluster-from-template/dialog/component.ts [hunk 3]
```diff
@@ -55,20 +55,21 @@ export class ClusterFromTemplateDialogComponent implements OnInit, OnDestroy {
...
 
-  get isValid(): boolean {
+  get valid(): boolean {
     return this._clusterTemplateService.isClusterStepValid;
   }
 
-  create(): void {
-    this._clusterTemplateService
+  get observable(): Observable<any> {
+    return this._clusterTemplateService
```
[the remark this answers is not in the document]
Comment (maciaszczykm) [reply src/app/shared/components/cluster-from-template/dialog/component.ts:63]: This has to have up do date values, so in most of the cases it cannot be done.

**Pull request merged**